messenger/src/messenger.rs

```diff
@@ -12,14 +16,85 @@ pub const BLOCK_STREAM: &str = "BLK";
 #[async_trait]
 pub trait Messenger: Sync + Send {
     async fn new(config: MessengerConfig) -> Result<Self, MessengerError>
-    where
-        Self: Sized;
-
+        where
+            Self: Sized;
+    fn messenger_type(&self) -> MessengerType;
     async fn add_stream(&mut self, stream_key: &'static str) -> Result<(), MessengerError>;
     async fn set_buffer_size(&mut self, stream_key: &'static str, max_buffer_size: usize);
     async fn send(&mut self, stream_key: &'static str, bytes: &[u8]) -> Result<(), MessengerError>;
     async fn recv(&mut self, stream_key: &'static str)
-        -> Result<Vec<(i64, &[u8])>, MessengerError>;
+                  -> Result<Vec<(i64, &[u8])>, MessengerError>;
 }
```

Comment:
Pulsar messenger has an `ack_message()` method.  Seems like we need to add this as part of the Messenger trait to have true interchangeability.  I guess right now ack_message() is just not being called in the plugin but probably should be.